### PR TITLE
ENH: Enable sklearn make_scorer

### LINKED ISSUE #173: scoring callbacks fail with sklearn make_scorer
Using a scoring callback with `make_scorer` fails:

```
from sklearn.metrics import accuracy_score, make_scorer

net = NeuralNetClassifier(..., callbacks=[('accuracy', BatchScoring(make_scorer(accuracy)))])
```
The reason is that the name of the `make_scorer` object cannot be determined. To mitigate this, pass the `name` argument to the scoring callback.

This bug is especially annoying as users might believe it is possible to pass an sklearn metric as scoring function, which fails with the prompt to use `make_scorer`, which will then fail with the error above.

### PR DESCRIPTION
**Pull request by thomasjpfan**
Fixes #173, #174

### CODE REVIEW

File: skorch/callbacks/scoring.py [hunk 1]
```diff
@@ -34,6 +35,19 @@ def cache_net_infer(net, use_caching, y_preds):
     del net.__dict__['infer']
 
 
+def convert_sklearn_metric_function(scoring):
+    """If ``scoring`` is a sklearn metric function, convert it to a
+    sklearn scorer and return it. Otherwise, return ``scoring`` unchanged."""
+    if callable(scoring):
+        module = getattr(scoring, '__module__', None)
+        if (hasattr(module, 'startswith') and
```
Comment (BenjaminBossan) [inline skorch/callbacks/scoring.py]: Could you please change the indentation of the conditions to increase readability:

```
if (
        cond0 and
        cond1 and ...
):
    return ...
```

File: skorch/callbacks/scoring.py [hunk 2]
```diff
@@ -64,10 +78,13 @@ def _get_name(self):
             return self.scoring
         if isinstance(self.scoring, partial):
             return self.scoring.func.__name__
+        if isinstance(self.scoring, _BaseScorer):
+            return self.scoring._score_func.__name__
         return self.scoring.__name__
 
     def initialize(self):
         self.best_score_ = np.inf if self.lower_is_better else -np.inf
+        self.scoring = convert_sklearn_metric_function(self.scoring)
```
Comment (BenjaminBossan) [inline skorch/callbacks/scoring.py]: Could you change the new variable to `scoring_`, so that the original `scoring` remains untouched?

**Comment by BenjaminBossan** [comment]
Thanks for the contribution. It looks very good to me, just 2 minor comments. I will be back in office on Monday and then give it a final review.

**Comment by benjamin-work** [comment]
TY
